server.go

```diff
@@ -294,6 +343,38 @@ func (s *Server) ReadMetricSocket(packetPool *sync.Pool, reuseport bool) {
 	}
 }
 
+// ReadTraceSocket listens for available packets to handle.
+func (s *Server) ReadTraceSocket(packetPool *sync.Pool, reuseport bool) {
+	// TODO This is duplicated from ReadMetricSocket and feels like it could be it's
+	// own function?
+
+	if s.TraceAddr == nil {
+		log.WithField("s.TraceAddr", s.TraceAddr).Fatal("Cannot listen on nil trace address")
+	}
+
+	serverConn, err := NewSocket(s.TraceAddr, s.RcvbufBytes, reuseport)
+	if err != nil {
+		// if any goroutine fails to create the socket, we can't really
+		// recover, so we just blow up
+		// this probably indicates a systemic issue, eg lack of
+		// SO_REUSEPORT support
+		log.WithError(err).Fatal("Error listening for UDP traces")
+	}
+	log.WithField("address", s.TraceAddr).Info("Listening for UDP traces")
+
+	for {
+		buf := packetPool.Get().([]byte)
+		n, _, err := serverConn.ReadFrom(buf)
+		if err != nil {
+			log.WithError(err).Error("Error reading from UDP trace socket")
+			continue
+		}
+
+		s.HandleTracePacket(buf[:n])
+		packetPool.Put(buf)
```

Comment:
remember to reslice the buffer back to its capacity `buf[:cap(buf)]`, otherwise its length will monotonically decrease each time you put it into the pool, until eventually it's empty